### PR TITLE
Use Map for OpcodeList and opcode handlers

### PR DESCRIPTION
**Pull request by ryanio**
This PR targets resolving the following from #775:

> * The opcodes handlers map is indexed by string, it should be done by number.
> * The opcodes handlers map and the opcodes OpcodeList's one, should probably be Maps. The semantic of Map is way simpler than an accessing an object with object[key].

### CODE REVIEW

**Comment by alcuadrado** [comment]
I have another idea of how this can be optimized. I built a small script simulating the current implementation, and this other idea, but somehow lost it 😅

I spent some time reading about how to create an efficient jump table in js, and apparently the best way of doing it is with an array. Something important is that we shouldn't leave empty indexes. Instead, an invalid/missing-opcode should be used. This way v8 will, hopefully, optimize it.

Another important thing can be to stop creating opcodes per hardfork, and put the HF validation logic inside each opcode. This way we can use a static array of ocpdes, increasing the chances of this getting optimized.

**Comment by jochem-brouwer** [comment]
I don't think we should go back to the point where we validate if an Op can be ran inside the Op itself. We can create multiple static arrays, where each array would be the array which holds the Ops of a certain HF. While checking HFs should not take a lot of time, I don't think we should check those at runtime but do that in the initialization phase.

File: packages/vm/lib/evm/interpreter.ts [hunk 1]
```diff
@@ -157,16 +157,15 @@ export default class Interpreter {
...
 
   /**
    * Get info for an opcode from VM's list of opcodes.
    */
   lookupOpInfo(op: number): Opcode {
-    const opcode = this._vm._opcodes[op] ? this._vm._opcodes[op] : this._vm._opcodes[0xfe]
-
-    return opcode
+    // if not found, return 0xfe: INVALID
+    return this._vm._opcodes.get(op) || this._vm._opcodes.get(0xfe)
```
Comment (evertonfraga) [inline packages/vm/lib/evm/interpreter.ts:168]: 👍

**Review by evertonfraga — APPROVED** [review]
That's great! 

Now, I guess the only thing is to decide along with @alcuadrado what route to take wrt Map vs. Arrays.

**Comment by ryanio** [comment]
thanks for the review @evertonfraga :) @alcuadrado said we can merge this for now since it's already an improvement, and do a follow-up PR for the array optimization.

i just had to resolve a merge conflict, but should be ready for a final approve and merge now.

**Pull request merged**